bazel/v8.patch

```diff
@@ -106,3 +107,107 @@ index c3768b8..d4a1dda 100644
  import pdl
 
  try:
+diff --git a/BUILD.bazel b/BUILD.bazel
+index 4e89f90e7e3..5d6c6d9bf3c 100644
+--- a/BUILD.bazel
++++ b/BUILD.bazel
```

Comment:
Can you upstream this support? Patches are hard to maintain and cause us to fall behind on external dependency updates due to friction. They are also problematic at security updates.

Reply:
Thanks for the feedback, @htuch .
One reason I needed to modify `v8.patch` was, wasm-c-api doesn't directly have a way of exposing `v8::isolate`, which is needed for vtune event handler (https://github.com/WebAssembly/wasm-c-api/issues/178)
Typically, other embedders of v8 like node and d8, use v8's embedder API, which expose the isolate, making it easier to add `vTune::GetVtuneCodeEventHandler`, but Envoy cannot do that, as it uses wasm-c-api.
I did not upstream these changes to v8 as I wasn't sure if it would be envoy-specific code, hence put it in `v8.patch`. I can take further look at it if this blocks the PR though.

Reply:
Can you file an issue with them exploring this option and share it here? It will be more maintainable in the long term if we don't grow the patch with this.

Reply:
https://bugs.chromium.org/p/v8/issues/detail?id=14056&q=bazel%20vtune&can=2
I opened an issue on v8, and I believe there are no objections for that contribution, so I will go ahead with it.
After contributing the patch, I will still need to have some code in `v8.patch` here for controlling the VTune capability via an envoy build flag, similar to this https://github.com/envoyproxy/envoy/issues/27326#issuecomment-1545947630
I hope that is alright?
Another question - will this PR be on hold till a new v8 with above changes is released, and envoy updates its v8 version? Or does it make sense to have the longer `v8.patch` now, and change it once a newer release v8 is available? Thanks.

Reply:
Probably OK if it's a very small patch (would be ideal to do via Bazel native mechanisms if possible though).

You reference the latest hash on `main` with a TODO to move to release version in the interim and we can land that commit.

Reply:
The relevant [v8 patch](https://chromium.googlesource.com/v8/v8.git/+/cc49dce9f5cb14daee8cc0946e996ba999f97d29) has recently been merged to v8's main branch.

> You reference the latest hash on main with a TODO to move to release version in the interim and we can land that commit.

Can you please clarify this? Did you mean that I should update envoy's `repository_locations.bzl` to use v8's main branch for now until there is a newer release? Or have the longer `v8.patch` for now with a comment stating that it will be removed once there is a newer v8 available? Thanks.

Reply:
>  Did you mean that I should update envoy's repository_locations.bzl to use v8's main branch for now until there is a newer release? 

Yep, see the examples like https://github.com/envoyproxy/envoy/blob/c53f81a43f44c7a3ea56bdda4a82044dbd78aafe/bazel/repository_locations.bzl#L580.